artcommon/artcommonlib/konflux/konflux_build_record.py

```diff
@@ -148,18 +116,73 @@ def get_nvr(self):
         self.nvr = f'{self.name}-{self.version}-{self.release}'
         return self.nvr
 
+    def __str__(self):
+        return json.dumps(self.to_dict(), indent=4)
+
+    def to_dict(self) -> dict:
+        """
+        Returns a dict object representing the build data.
+        KonfluxDb.add_builds() can use data in this form to store the build info into the DB.
+        """
+
+        return {
+            key: (value.strftime("%Y-%m-%dT%H:%M:%S") if isinstance(value, datetime)
+                  else value.value if isinstance(value, Enum) else value)
+            for key, value in self.__dict__.items()
+        }
+
+
+class KonfluxBuildRecord(KonfluxRecord):
+    TABLE_ID = constants.BUILDS_TABLE_ID
 
-def from_result_row(row: Row) -> typing.Optional[KonfluxBuildRecord]:
     """
-    Given a google.cloud.bigquery.table.Row object, construct and return a KonfluxBuild object
+    This class represents the data structure we use to store build information in Konflux
+    By calling the konfluxdb.support.konflux_build.generate_build_schema() function,
+    a table schema can be auto-generated from this class definition.
+    The schema we get can then be injected into the konfluxdb.support.konflux_db.SupportKonfluxDb.create_table()
+    function, that will create the table in BigQuery for us.
+
+    In order for this to work, this class should contain no fields other than those we want to appear as table columns.
     """
 
-    try:
-        return KonfluxBuildRecord(**{
-            field: (row[field])
-            for field in row.keys()
-        })
+    def __init__(self, name: str = '', group: str = '', version: str = '', release: str = '', assembly: str = '',
+                 el_target: str = '', arches: list = [], installed_packages: list = [], parent_images: list = [],
+                 source_repo: str = '', commitish: str = '', rebase_repo_url: str = '', rebase_commitish: str = '',
+                 embargoed: bool = False, start_time: datetime = None, end_time: datetime = None,
+                 artifact_type: ArtifactType = ArtifactType.IMAGE, engine: Engine = Engine.KONFLUX,
+                 image_pullspec: str = '', image_tag: str = '',
+                 outcome: KonfluxBuildOutcome = KonfluxBuildOutcome.SUCCESS, art_job_url: str = '',
```

Comment:
I think we had discussed about `PENDING`. This can be changed without updates to the DB (it's a simple string field), so let's change it when actually needed. This means, when we start storing actual Konflux pipeline build records.